File: api/src/main/java/environment/map/GameMapLayer.java
package environment.map;

import system.objects.ResourceObject;
import utils.FrogException;

public class GameMapLayer implements ResourceObject
{
	private Integer[][] tiles;
	private int width, height;
	
	public GameMapLayer(int width, int height)
	{
		this.tiles = new Integer[height][width];
		this.width = width;
		this.height = height;
	}
	
	public void load()
	{
		/*for(int x = 0; x < this.width; x++)
		{
			for(int y = 0; y < this.height; y++)
			{
				// Load Tile (x, y)
			}
		}*/
	}
	
	public void unload()
	{
		
	}
	
	/**
	 * Obtient la valeur de la tuile à la position donnée
	 * @param relativeX La valeur horizontale relative au chunk de la position
	 * @param relativeY La valeur verticale relative au chunk de la position
	 * @return La valeur de la tuile à la position donnée, null si inexistant
	 */
	public Integer getTile(int relativeX, int relativeY)
	{
		if(relativeX >= 0 && relativeX < this.width && relativeY >= 0 && relativeY < this.height)
			return this.tiles[relativeY][relativeX];
		return null;
	}
	
	/**
	 * Met à jour la valeur de la tuile à la position donnée
	 * @param relativeX La valeur horizontale relative au chunk de la position donnée
	 * @param relativeY La valeur verticale relative au chunk de la position donnée
	 * @param value La nouvelle valeur de la tuile
	 * @throws FrogException Exception jetée lorsque la position donnée est située en dehors des limites définies
	 */
	public void setTile(int relativeX, int relativeY, Integer value) throws FrogException
	{
		if(relativeX >= 0 && relativeX < this.width && relativeY >= 0 && relativeY < this.height)
			this.tiles[relativeY][relativeX] = value;
		else
			throw new FrogException("Tentative de modification d'une tuile en dehors des limites définies");
	}
}

File: godot/scripts/api/environment/map/GameMapLayer.cs
using FrogCreator.Api.System.Objects;
using FrogCreator.Api.Utils;

namespace FrogCreator.Api.Environment.Map;

public class GameMapLayer : IResource
{
    private int?[,] _tiles;
    private int _width, _height;

    public GameMapLayer(int width, int height)
    {
        _tiles = new int?[height, width];
        _width = width;
        _height = height;
    }

    public void Load()
    {
        /*for(int x = 0; x < _width; x++)
        {
            for(int y = 0; y < _height; y++)
            {
                // Load Tile (x, y)
            }
        }*/
    }

    public void Unload()
    {
    }

    /// <summary>
    /// Obtient la valeur de la tuile à la position donnée
    /// </summary>
    /// <param name="relativeX">La valeur horizontale relative au chunk de la position</param>
    /// <param name="relativeY">La valeur verticale relative au chunk de la position</param>
    /// <returns>La valeur de la tuile à la position donnée, null si inexistant</returns>
    public int? GetTile(int relativeX, int relativeY)
    {
        if (relativeX >= 0 && relativeX < _width && relativeY >= 0 && relativeY < _height)
            return _tiles[relativeY, relativeX];
        return null;
    }

    /// <summary>
    /// Met à jour la valeur de la tuile à la position donnée
    /// </summary>
    /// <param name="relativeX">La valeur horizontale relative au chunk de la position donnée</param>
    /// <param name="relativeY">La valeur verticale relative au chunk de la position donnée</param>
    /// <param name="value">La nouvelle valeur de la tuile</param>
    /// <exception cref="FrogException">Exception jetée lorsque la position donnée est située en dehors des limites définies</exception>
    public void SetTile(int relativeX, int relativeY, int value)
    {
        if (relativeX >= 0 && relativeX < _width && relativeY >= 0 && relativeY < _height)
            _tiles[relativeY, relativeX] = value;
        else
            throw new FrogException("Tentative de modification d'une tuile en dehors des limites définies");
    }
}
